File: src/main/scala/checker/constraints/incremental/Push.java
package checker.constraints.incremental;


import java.util.Set;

public class Push extends BranchOp {
    public Push(Set<Integer>[] domains){super(domains);}
    @Override
    public BranchOp clone(){
        return new Push(domains.clone());
    }
    @Override
    public String toString() {
        return "Push\n";
    }
}

File: src/main/scala/checker/incremental/Push.scala
package checker.incremental

/**
  * branching operation representing a push operation.
  * It means that the current state should be pushed
  * on the solver's trail.
  *
  * @param doms : 'should be' values of the domains
  *             before the push branching operation.
  */
class Push(val doms: Array[Set[Int]]) extends BranchOp(doms) {
  override def clone = new Push(domains.clone)

  override def toString = "Push"
}

